hal/api/critical.h

```diff
@@ -24,6 +24,18 @@
 extern "C" {
 #endif
 
+
+/** Determine the current interrupts disabled state
+  *
+  * This function can be called to determine whether or not interrupts are currently disabled.
+  * \note
+  * NOTE:
+  * This function works for both cortex-A and cortex-M, although the underlyng implementation
+  * differs.
+  * @return true if interrupts are disabled, false otherwise
+  */
+bool get_interrupts_disabled(void);
```

Comment:
Happy to switch the logic from the original change although technically that was not a change that was part of this PR.